rules/mockery-to-prophecy/src/Rector/MethodCall/MockeryCreateMockToProphizeRector.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\MockeryToProphecy\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Type\ObjectType;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * @see \Rector\MockeryToProphecy\Tests\Rector\Class_\MockeryToProphecyRector\MockeryToProphecyRectorTest
+ */
+final class MockeryCreateMockToProphizeRector extends AbstractRector
+{
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [StaticCall::class, MethodCall::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if ($node instanceof MethodCall) {
+            foreach ($node->args as $position => $arg) {
+                /** @var Node\Expr\Variable $value */
+                $value = $arg->value;
+                if ($value instanceof Node\Expr\Variable) {
+                    /** @var Scope $scope */
+                    $scope = $value->getAttribute('PHPStan\Analyser\Scope');
+                    /** @var ObjectType */
+                    $type = $scope->getVariableType($value->name);
+
+                    if ($type->getClassName() === 'Mockery\MockInterface') {
+                        $node->args[$position] = $this->createArg($this->createMethodCall($arg->value, 'reveal'));
+                    }
+                }
+            }
+        }
```

Comment:
If these is too much logic at once, I usually extract per-node methods, e.g.  `refactorClassMethod()`

And `refactorStaticCall()` for the next method.